backend/api_handler.py

```diff
@@ -163,6 +148,41 @@ def handler(event, context):
             'schema': SCHEMA,
         }
 
+        query = json.loads(event.get('body'))
+
+        # Logic to block when in maintenance
+        # Check if in some maintenance mode
+        # Check if in maintenance status is not INACTIVE
+        # Check if the user belongs to a 'DAAdministrators' group
+        if config.get_property('modules.maintenance.active'):
```

Comment:
Suggestion:  this whole if block might be simplified like the following, create vars for each of the calls, it is a bit more readable this way as well.

<img width="884" alt="Screenshot 2024-05-03 at 4 06 50 PM" src="https://github.com/data-dot-all/dataall/assets/97474536/9882d847-9b64-4f9e-b73c-f8c68d8fa06f">

`
maintenance_active = config.get_property('modules.maintenance.active')
maintenance_window_mode = MaintenanceService._get_maintenance_window_mode(engine=ENGINE)
maintenance_window_status = MaintenanceService.get_maintenance_window_status(engine=ENGINE).status
is_tenant_admin = TenantPolicyValidationService.is_tenant_admin(groups)

if maintenance_active and maintenance_window_status != MaintenanceStatus.INACTIVE.value and not is_tenant_admin:
    if maintenance_window_mode == MaintenanceModes.NOACCESS.value and query.get('operationName', '') not in MAINTENANCE_ALLOWED_OPERATIONS:
        return send_unauthorized_response(
            query=query,
            message='Access Restricted: data.all is currently undergoing maintenance, and your actions are temporarily blocked.',
        )
    elif maintenance_window_mode == MaintenanceModes.READONLY.value and query.get('query', '').split()[0] == 'mutation':
        return send_unauthorized_response(
            query=query,
            message='Access Restricted: data.all is currently undergoing maintenance, and your actions are temporarily blocked.',
        )

`

Reply:
Thanks for the suggestion @rbernotas . Updated in the next commit